### DISCUSSION TITLE
Is it possible create async no store actions ?

### DISCUSSION THREAD

**Discussion by budarin** [pmndrs/zustand#1791] [Q&A] [answered]
Unfortunately, I could not find a description of how to create an asynchronous no store action :(

Is it possible or not?

**Comment by dai-shi** [comment]
what is an async no store action?

**Comment by budarin** [comment]
it's like async store actions but [no store actions](https://docs.pmnd.rs/zustand/guides/practice-with-no-store-actions)

```js
const useStore = create((set) => ({
  fishies: {},
  fetch: async (pond) => {
    const response = await fetch(pond)
    set({ fishies: await response.json() })
  },
}))
```
how to write no store action for the fetch action above?

I need to have an ability to set and get data from the store and do some network requests in the no store action

For example: this store action to rewrite in no to store action: 

```js
  deleteTodo: async (id ) => {
    const todo = get.todos.byId[id];
    
    // optimistic update - delete todo local
    set({ ... })
    
    try {
      // try to delete todo 4 on a server
      const response = await fetch(..); 
      
      // delete todo on a server
      set({ fishies: await response.json() })
    } catch(e) {
      // restore todo
      set({ ... })
    }
  }
```

**Reply by dai-shi** [reply]
> how to write no store action for the fetch action above?

Something like this:
```js
const useFishStore = create(() => ({
  fishies: {},
}))

const fetchFish = async (pond) => {
  const response = await fetch(pond)
  useFishStore.setState({ fishies: await response.json() })
}
```

**Reply by budarin** [reply]
what about to have an ability to get state in fetchFish?

**Reply by dai-shi** [reply, marked as answer]
`useFishStore.getState()` will do.
